akka-http2-support/src/main/scala/akka/http/impl/engine/http2/FrameParser.scala

```diff
@@ -49,6 +49,11 @@ class FrameParser(shouldReadPreface: Boolean) extends ByteStringParser[FrameEven
   def parseFrame(tpe: FrameType, flags: ByteFlag, streamId: Int, payload: ByteReader): FrameEvent = {
     // TODO: add @switch? seems non-trivial for now
     tpe match {
+      case GOAWAY ⇒
+        Http2Compliance.requireZeroStreamId(streamId)
```

Comment:
Ok, that's probably ok; and we'd handle the 

> The GOAWAY frame applies to the connection, not a specific stream. An endpoint MUST treat a GOAWAY frame with a stream identifier other than 0x0 as a connection error (Section 5.4.1) of type PROTOCOL_ERROR.

on the error handling level, though perhaps specific exception that it was a GOAWAY would be better then?